src/vs/workbench/services/workingCopy/test/browser/untitledScratchpadWorkingCopy.test.ts

```diff
@@ -0,0 +1,273 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as assert from 'assert';
+import { VSBufferReadableStream, VSBuffer, streamToBuffer, bufferToStream, readableToBuffer, VSBufferReadable } from 'vs/base/common/buffer';
+import { CancellationToken } from 'vs/base/common/cancellation';
+import { DisposableStore } from 'vs/base/common/lifecycle';
+import { Schemas } from 'vs/base/common/network';
+import { basename } from 'vs/base/common/resources';
+import { consumeReadable, consumeStream, isReadable, isReadableStream } from 'vs/base/common/stream';
+import { URI } from 'vs/base/common/uri';
+import { IInstantiationService } from 'vs/platform/instantiation/common/instantiation';
+import { IUntitledFileWorkingCopyModelFactory, UntitledFileWorkingCopy } from 'vs/workbench/services/workingCopy/common/untitledFileWorkingCopy';
+import { TestUntitledFileWorkingCopyModel } from 'vs/workbench/services/workingCopy/test/browser/untitledFileWorkingCopy.test';
+import { TestServiceAccessor, workbenchInstantiationService } from 'vs/workbench/test/browser/workbenchTestServices';
+
+export class TestUntitledFileWorkingCopyModelFactory implements IUntitledFileWorkingCopyModelFactory<TestUntitledFileWorkingCopyModel> {
+
+	async createModel(resource: URI, contents: VSBufferReadableStream, token: CancellationToken): Promise<TestUntitledFileWorkingCopyModel> {
+		return new TestUntitledFileWorkingCopyModel(resource, (await streamToBuffer(contents)).toString());
+	}
+}
+
+suite('UntitledScratchpadWorkingCopy', () => {
```

Comment:
copied from UntitledFileWorkingCopy suite with changes to account for scratchpad behavior.

Reply:
You could try to reuse the same code in both suites with a parameter for using scratchpad or not.

Reply:
each test would have its own extracted function right? that feels pretty strange. I was hoping to just do a data driven test, but that's not yet supported with the test runner (but might be soon https://github.com/microsoft/vscode/issues/182788).

Reply:
Yeah not a big deal. We do this kind of shared tests when we want to execute the exact same tests in web and desktop, but if the tests are differing, maybe not worth the efford.